### PR TITLE
Support hot-reloading built-in connectors when the config file changes

### LINKED ISSUE #602: Hot-reload builtin connectors when connector config file changes
The builtin connectors are maintained in a `TreeMap` in the `MeshConnectorsManager`. 
To load the list of connectors into the map, `searchForConnectors` is called. Currently, this method is only called:
a) when the worker service starts
b) when `MeshComponentImpl#reloadConnectors` is called through the API

To be able to change the list of supported connectors without having to restart the broker, we should watch the `conf/connectors.yaml` file for changes and call `searchForConnectors` whenever we detect a change.

The solution should take into consideration that the `connectors.yaml` file might be mounted through a k8s `ConfigMap` (see issue with watching bind-mounted files here: https://blog.arkey.fr/2019/09/13/watchservice-and-bind-mount/)

### PR DESCRIPTION
**Pull request by alpreu**
Closes #602. 

Detecting the change in the config file is achieved through a polling solution. The reason for this is that with sym-linked files coming from a ConfigMap, the [WatchService approach fails](https://blog.arkey.fr/2019/09/13/watchservice-and-bind-mount/), and detecting change events directly from Kubernetes would require additional dependencies.

### CODE REVIEW

File: src/main/java/io/functionmesh/compute/models/MeshWorkerServiceCustomConfig.java [hunk 1]
```diff
@@ -239,6 +239,16 @@ public class MeshWorkerServiceCustomConfig {
     )
     protected boolean enableTrustedMode = false;
 
+    @FieldContext(
+            doc = "The interval in which the mesh worker service reloads the built-in connectors. By default it is 60."
+    )
+    protected long connectorSearchIntervalSeconds = 60;
```
Comment (alpreu) [inline src/main/java/io/functionmesh/compute/models/MeshWorkerServiceCustomConfig.java:245]: I set this to 1 minute for now, but I'm open to discuss what a good default could be

File: src/main/java/io/functionmesh/compute/worker/MeshConnectorsManager.java [hunk 2]
```diff
@@ -72,7 +95,7 @@ public FunctionMeshConnectorDefinition getConnectorDefinition(String connectorTy
     }
 
     public void reloadConnectors() {
-        connectors = searchForConnectors();
+        searchExecutor.submit(this::loadConnectorsIfNeeded);
```
Comment (alpreu) [inline src/main/java/io/functionmesh/compute/worker/MeshConnectorsManager.java:98]: Do we need to guard here against other threads? I'm not sure

**Review by nlu90 — COMMENTED** [review]
Can we just read from the config file whenever the `get connectors` rest API is called?

This way, we avoid introducing the extra check-and-reload thread and reduce the code complexity.

The response latency may be increased. But given `get built-in connectors` is a low-frequency request and the disk latency is around ms level. That should be acceptable.

**Comment by alpreu** [comment]
> Can we just read from the config file whenever the `get connectors` rest API is called?
> 
> This way, we avoid introducing the extra check-and-reload thread and reduce the code complexity.
> 
> The response latency may be increased. But given `get built-in connectors` is a low-frequency request and the disk latency is around ms level. That should be acceptable.

@nlu90 That is the current way, and the problem I see with this approach is that it would require users to always call `get built-in connector` before `create` or `update`

**Review by nlu90 — APPROVED** [review]

**Review by jiangpengcheng — APPROVED** [review]

**Review by freeznet — APPROVED** [review]

**Pull request merged**